### DISCUSSION TITLE
Method to reset the { offset: [x, y] } or first gesture.

### DISCUSSION THREAD

**Discussion by wileyknight** [pmndrs/use-gesture#288] [Q&A]
I am creating an interaction in three.js and it would be helpful to be able to reset the offset or first gesture when it crosses a threshold.

Example:
```
const bind = useGesture(
    {
        onDrag: ({ offset: [x, y] }) => {
            if (y > somevalue) {
                state.offset[1] = 0;
            }
        }
    }
);
```

I need to know the distance from the initial gesture but I also want to reset it as a new starting point when I get to a new "area" or even click a button or some other interaction. I can do this on my own but it would be pretty convenient if there is already a method.

Thank you.

**Comment by dbismut** [comment]
I think it's fairly easy to do this in userland, this would add extra logic to the lib. Although the handler could export a `set` function allowing to write to the state imperatively. But I suspect this would generate some dangerous side effects.

**Reply by ElectricPulse** [reply]
This cannot be done in user land since usePinch when emulating a mouse wheel ouputs different .delta values based on the current offset. So ignoring offset and just looking at delta, causes the delta values to almost exponentially grow. Moreover things like scaleBounds and rubberbanding stop working.
